src/snowflake/snowpark/_functions/general_functions.py

```diff
@@ -0,0 +1,213 @@
+        |     |     |     |      |                    |            |]     |                   |     |
+        ---------------------------------------------------------------------------------------------
+        <BLANKLINE>
+    """
+
+    if _emit_ast:
+        ast = proto.Expr()
+        if datatype is None:
+            build_builtin_fn_apply(ast, "lit", literal)
+        else:
+            build_builtin_fn_apply(ast, "lit", literal, datatype)
+
+        if isinstance(literal, Column):
+            # Create new Column, and assign expression of current Column object.
+            # This will encode AST correctly.
+            c = Column("", _emit_ast=False)
+            c._expression = literal._expression
+            c._ast = ast
+            return c
+        return Column(Literal(literal, datatype=datatype), _ast=ast, _emit_ast=True)
+
+    if isinstance(literal, Column):
+        return literal
+
+    return Column(Literal(literal, datatype=datatype), _ast=None, _emit_ast=False)
+
+
+def _call_function(
+    name: str,
+    *args: ColumnOrLiteral,
+    is_distinct: bool = False,
+    api_call_source: Optional[str] = None,
+    is_data_generator: bool = False,
+    _ast: proto.Expr = None,
+    _emit_ast: bool = True,
+) -> Column:
+
+    if _emit_ast and _ast is None:
+        _ast = build_function_expr(name, args)
+
+    args_list = parse_positional_args_to_list(*args)
+    expressions = [Column._to_expr(arg) for arg in args_list]
+    return Column(
+        FunctionExpression(
+            name,
+            expressions,
+            is_distinct=is_distinct,
+            api_call_source=api_call_source,
+            is_data_generator=is_data_generator,
+        ),
+        _ast=_ast,
+        _emit_ast=_emit_ast,
+    )
+
+
+def call_function(
+    function_name: str,
+    *args: ColumnOrLiteral,
+    _emit_ast: bool = True,
+) -> Column:
+    """Invokes a Snowflake `system-defined function <https://docs.snowflake.com/en/sql-reference-functions.html>`_ (built-in function) with the specified name
+    and arguments.
+
+    Args:
+        function_name: The name of built-in function in Snowflake
+        args: Arguments can be in two types:
+
+            - :class:`~snowflake.snowpark.Column`, or
+            - Basic Python types, which are converted to Snowpark literals.
+
+    Example::
+        >>> df = session.create_dataframe([1, 2, 3, 4], schema=["a"])  # a single column with 4 rows
+        >>> df.select(call_function("avg", col("a"))).show()
+        ----------------
+        |"AVG(""A"")"  |
+        ----------------
+        |2.500000      |
+        ----------------
+        <BLANKLINE>
+
+    """
+    ast = (
+        build_function_expr("call_function", [function_name, *args])
+        if _emit_ast
+        else None
+    )
+    return _call_function(function_name, *args, _ast=ast, _emit_ast=_emit_ast)
+
+
+def function(function_name: str, _emit_ast: bool = True) -> Callable:
+    """
```

Comment:
can we remove the docstring that are duplicated here and just maintain the docstrings in `functions.py`